scalableminds/webknossos
# Add context menu for nodes

This PR adds a context menu for nodes that offers various actions.

Currently, the node context menu is only supported on nodes.
I could also change it, so that it is possible to open it up anywhere, no matter whether there is a node or not.
The following operations are currently implemented and are available if the menu is opened on an existing node:
- One of the following:
    - Create Edge To This Node (When the active node and the clicked node are in the same tree)
    - Create Edge & Merge With This Tree (When the active tree and the tree the clicked node is in are not the same)
- Delete Edge To This Node (Only enabled, when the active node is directly connected to this node)
- Measure Length To This Node (Only enabled, when the active node and the clicked node are in the same tree)
- Measure Length Of This Tree

Additional useful feature suggestions are welcome :smile_cat: 


### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Open a skeleton tracing and create some trees.
- The context menu should open with shift + right-click and should automatically close when an option is clicked or the user clicks somewhere else.
    - Currently, firefox still opens up the usual context menu which is kinda bothersome. I'll try to work out a fix for this.
- Each option in the context menu should do what is says and should only be enabled if the action makes sense and is possible.
- Play around with the options and try out edge cases like deleting a node in a "branch" should split the current tree.

### Issues:
- fixes #4906

------
